### PR TITLE
Fix #896

### LINKED ISSUE #896: 0.16 breaks "external" filters
In 0.15.7, one could use filters that are available in the the HDF5 plugin directory (even if they were not previously known or supported by `HDF5.jl`) by passing `filter=(FILTER_ID, FLAGS, cd_values...)` as a keyword to `create_dataset`.  In 0.16.1, this leads to a would-be infinite loop:
```
ERROR: LoadError: StackOverflowError:
Stacktrace:
      [1] append!
        @ ~/.julia/dev/HDF5/src/filters/filters.jl:292 [inlined]
      [2] push!(a::HDF5.Filters.FilterPipeline{HDF5.DatasetCreateProperties}, iter::Int64)
        @ Base ./array.jl:986
--- the last 2 lines are repeated 79982 more times ---
 [159967] append!
        @ ~/.julia/dev/HDF5/src/filters/filters.jl:292 [inlined]
```
Is there a new way to work with these external filters in v0.16.x?

### PR DESCRIPTION
**Pull request by mkitti**
Create additional methods to interpret a `Tuple` as an `UnknownFilter`.

### CODE REVIEW

File: src/filters/filters.jl [hunk 1]
```diff
@@ -287,12 +291,15 @@ function Base.delete!(filters::FilterPipeline, ::Type{F}) where {F<:Filter}
...
     return filters
 end
-function Base.append!(filters::FilterPipeline, extra)
+function Base.append!(filters::FilterPipeline, extra::Union{AbstractVector{<:Filter}, NTuple{N, Filter} where N})
     for filter in extra
         push!(filters, filter)
     end
     return filters
 end
+function Base.append!(filters::FilterPipeline, extra::NTuple{N, Integer}) where N
```
Comment (musm) [inline src/filters/filters.jl:300]: I guess this is only added to support existing use cases.

Would it make sense to wrap this is a deprecation warning and force users to manually wrap things in `UnkownFilter`.

I'm leaning towards not doing that and just leaving this as in the PR?  Just a thought.

Comment (mkitti) [reply src/filters/filters.jl:300]: We have already deprecated the upstream interface, so I think that warning should be sufficient. Keeping this just prevents the rather nasty stack overflow situation.

Comment (musm) [reply src/filters/filters.jl:300]: This should've been in reference to the addition
```
function Base.push!(p::FilterPipeline, f::Tuple)
```

Comment (mkitti) [reply src/filters/filters.jl]: I deprecated all the `Tuple` based methods. The tests give the following warnings.

```
┌ Warning: `filter=(307, 0)` keyword option is deprecated, use `filters=(307, 0)` instead
│   caller = setproperty!(p::HDF5.DatasetCreateProperties, name::Symbol, val::Tuple{Int32, Int64}) at properties.jl:47
└ @ HDF5 C:\Users\kittisopikulm\.julia\dev\HDF5\src\properties.jl:47
┌ Warning: `append!(filters::FilterPipeline, extra::NTuple{N, Integer}) where N` is deprecated, use `append!(filters, [UnknownFilter(extra...)])` instead.
│   caller = set_filters! at properties.jl:403 [inlined]
└ @ Core C:\Users\kittisopikulm\.julia\dev\HDF5\src\properties.jl:403
```

**Comment by musm** [comment]
LGTM, could we finally move the deprecations to https://github.com/JuliaIO/HDF5.jl/blob/master/src/deprecated.jl  makes things a lot easier during major version updates. 

Thanks!

**Comment by mkitti** [comment]
Deprecated statements moved. This should be ready to go now.

**Comment by musm** [comment]
Could you enable edits from other users in the PR?

File: src/deprecated.jl [hunk 2]
```diff
@@ -53,6 +53,15 @@ function set_shuffle!(p::Properties, ::Tuple{})
...
 end
 
+### Fix Issue #896
+import Base: append!, push!
+import .Filters: UnknownFilter
+
+@deprecate append!(filters::Filters.FilterPipeline, extra::NTuple{N, Integer}) where N append!(filters, [UnknownFilter(extra...)])
+@deprecate push!(p::Filters.FilterPipeline, f::NTuple{N, Integer}) where N push!(p, UnknownFilter(f...))
+@deprecate UnknownFilter(t::Tuple) UnknownFilter(t...)
+# End Fix Issue #896
```
Comment (musm) [inline src/deprecated.jl:63]:
```suggestion
```

File: src/deprecated.jl [hunk 3]
```diff
@@ -53,6 +53,15 @@ function set_shuffle!(p::Properties, ::Tuple{})
     set_shuffle!(p, true)
 end
 
+### Fix Issue #896
+import Base: append!, push!
+import .Filters: UnknownFilter
+
```
Comment (musm) [inline src/deprecated.jl:59]:
```suggestion
```

File: src/deprecated.jl [hunk 4]
```diff
@@ -53,6 +53,15 @@ function set_shuffle!(p::Properties, ::Tuple{})
     set_shuffle!(p, true)
 end
 
+### Fix Issue #896
```
Comment (musm) [inline src/deprecated.jl:56]:
```suggestion
### Changed in PR #902
```

**Review by musm — APPROVED** [review]

**Comment by mkitti** [comment]
It is coming from a fork in another org, so I'm not sure I can do it directly.. But you should still have access to that repository.

**Pull request merged**